sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -335,6 +343,41 @@ case class In(value: Expression, list: Seq[Expression]) extends Predicate {
        """.stripMargin)
   }
 
+  private def genCodeWithSwitch(ctx: CodegenContext, ev: ExprCode): ExprCode = {
+    val (nullLiterals, nonNullLiterals) = list.partition {
+      case Literal(null, _) => true
+      case _ => false
+    }
+    val listGen = nonNullLiterals.map(_.genCode(ctx))
+    val valueGen = value.genCode(ctx)
+
+    val caseBranches = listGen.map(literal =>
+      s"""
+         |case ${literal.value}:
+         |  ${ev.value} = true;
+         |  break;
+       """.stripMargin)
+
+    ev.copy(code =
+      code"""
+         |${valueGen.code}
+         |${CodeGenerator.JAVA_BOOLEAN} ${ev.isNull} = ${valueGen.isNull};
+         |${CodeGenerator.JAVA_BOOLEAN} ${ev.value} = false;
+         |if (!${valueGen.isNull}) {
+         |  switch (${valueGen.value}) {
+         |    ${caseBranches.mkString("")}
```

Comment:
we should consider that if the number of items is very big, this can cause a compile exception due to the method size limit. So we should use the proper splitting methods for the code

Reply:
@aokolnychyi Could you please address @mgaido91 's comment? The current code will throw an exception for a huge sequence of `In`.

Reply:
Could you please add test cases that could cause more than 64KB Java bytecode size in one switch statement?